roboteam_ai/src/skills/Attack.cpp

```diff
@@ -18,21 +18,49 @@ void Attack::onInitialize() {
 }
 
 /// Get an update on the skill
+
 bt::Node::Status Attack::onUpdate() {
     if (! robot) return Status::Running;
 
     if (shot && !World::botHasBall(robot->id, true)) {
         return Status::Success;
     }
 
+
+    ballTarget = Field::get_their_goal_center();
+    genevaState = 3;
+
+    if (properties->hasInt("genevaState")) {
+        genevaState = properties->getInt("genevaState");
+    }
+
+    // Overwrite set genevaState if autoGeneva is true
+    if (properties->getBool("autoGeneva")) {
+        roboteam_msgs::GeometryFieldSize field = Field::get_field();
+        if (ball->pos.y > 0) {
+            genevaState = 5;
+            ballTarget = {field.field_length / 2, 0.33 * field.goal_width};
+        } else {
+            genevaState = 1;
+            ballTarget = {field.field_length / 2, -0.33 * field.goal_width};
+        }
+
+        // Set the geneva to one angle lower if the distance from the goal is more than 1,5 times the goal width
+        if (ball->pos.x < field.field_width - 1.5 * field.goal_width) {
+            genevaState += (3 - genevaState) / 2;
```

Comment:
this can get weird with rounding. Maybe use `ceil` and/or `floor`

Reply:
Yes